### ISSUE TITLE
Browsers compatibility issues: Files with [ ] or ( ) in the name do have incorrectly encoded hyperlink in "raw" link.

### ISSUE THREAD

**Issue by TomaszSzt** [gitblit/gitblit#1375]
Steps to reproduce
1.Create a repository
2.Push a file named like, for an example "Px-[My file title].txt"
3.Enter the GITBlit page with this file, regardless if through commit or through the tree.
4.Find a "Raw" link to that file. Observe it will be looking like:
../../raw/somethi.git/9533e7971509d8476a9845d8fc31707abee9248c/Px-[My file title].tx

Notice that all un-allowed URL characters are NOT escaped. This includes spaces and [ ] brackets.
Click it in Firefox or Edge.
Observe Server Error.
Notice that browsers DID escape correctly spaces but DID NOT escape brackets. I think they should, but they did not.

5.Manually escape all characters using %20 for space %5A for [ and %5D for ] in browser address bar and tap enter.
6.Observe that server correctly responds to that link.

Wokaround

The only workaround to get raw file with such content through web interface is to download zipped repository.

Recomendation

All hyperlinks generated by GITBLIT should be valid, correctly escaped URLs.

Best regards,
  Tomasz Sztejka

**Comment by flaix** [comment]
Yes, the raw link is not escaped, I just could not reproduce this problem, as the browsers do not mind. I will try this under Windows 10.

**Comment by flaix** [comment]
This seems to happen when both spaces and brackets are in a file name. I could not find a specific pattern or combination.
The browser escapes spaces everywhere. While file names with spaces without brackets reach the servlet unescaped, this is not true when there are also brackets in the path. I guess that Jetty has some weird rule about when to treat spaces how. Might be interesting to try this with Gitblit in a Tomcat.
